In [9]:
import pandas as pd
import numpy as np
import datetime as dt

In [11]:
df_cd_2010_to_2019 = pd.read_csv(r'C:\Documents\projects\HackLA\311\data\crime data\Crime_Data_from_2010_to_2019.csv')

In [12]:
df_cd_2010_to_2019.head()

DR_NO               Date Rptd                DATE OCC  TIME OCC  AREA   \
0    1307355  02/20/2010 12:00:00 AM  02/20/2010 12:00:00 AM      1350     13   
1   11401303  09/13/2010 12:00:00 AM  09/12/2010 12:00:00 AM        45     14   
2   70309629  08/09/2010 12:00:00 AM  08/09/2010 12:00:00 AM      1515     13   
3   90631215  01/05/2010 12:00:00 AM  01/05/2010 12:00:00 AM       150      6   
4  100100501  01/03/2010 12:00:00 AM  01/02/2010 12:00:00 AM      2100      1   

   AREA NAME  Rpt Dist No  Part 1-2  Crm Cd  \
0     Newton         1385         2     900   
1    Pacific         1485         2     740   
2     Newton         1324         2     946   
3  Hollywood          646         2     900   
4    Central          176         1     122   

                                         Crm Cd Desc  ... Status  \
0                           VIOLATION OF COURT ORDER  ...     AA   
1  VANDALISM - FELONY ($400 & OVER, ALL CHURCH VA...  ...     IC   
2                          OTHER MISCELLANEOUS CRIME  ...     IC   
3                           VIOLATION OF COURT ORDER  ...     IC   
4                                    RAPE, ATTEMPTED  ...     IC   

    Status Desc Crm Cd 1 Crm Cd 2  Crm Cd 3 Crm Cd 4  \
0  Adult Arrest    900.0      NaN       NaN      NaN   
1   Invest Cont    740.0      NaN       NaN      NaN   
2   Invest Cont    946.0      NaN       NaN      NaN   
3   Invest Cont    900.0    998.0       NaN      NaN   
4   Invest Cont    122.0      NaN       NaN      NaN   

                                  LOCATION                     Cross Street  \
0   300 E  GAGE                         AV                              NaN   
1          SEPULVEDA                    BL  MANCHESTER                   AV   
2  1300 E  21ST                         ST                              NaN   
3          CAHUENGA                     BL  HOLLYWOOD                    BL   
4          8TH                          ST  SAN PEDRO                    ST   

       LAT       LON  
0  33.9825 -118.2695  
1  33.9599 -118.3962  
2  34.0224 -118.2524  
3  34.1016 -118.3295  
4  34.0387 -118.2488  

[5 rows x 28 columns]

In [15]:
df_cd_2010_to_2019.shape

(2119797, 28)

In [16]:
df_cd_2010_to_2019.dtypes

DR_NO               int64
Date Rptd          object
DATE OCC           object
TIME OCC            int64
AREA                int64
AREA NAME          object
Rpt Dist No         int64
Part 1-2            int64
Crm Cd              int64
Crm Cd Desc        object
Mocodes            object
Vict Age            int64
Vict Sex           object
Vict Descent       object
Premis Cd         float64
Premis Desc        object
Weapon Used Cd    float64
Weapon Desc        object
Status             object
Status Desc        object
Crm Cd 1          float64
Crm Cd 2          float64
Crm Cd 3          float64
Crm Cd 4          float64
LOCATION           object
Cross Street       object
LAT               float64
LON               float64
dtype: object

In [13]:
df2019 = df_cd_2010_to_2019[df_cd_2010_to_2019['DATE OCC'].str.contains('02/20/2019')]

In [17]:
df_cd_2010_to_2019['Date Rptd'] = pd.to_datetime(df_cd_2010_to_2019['Date Rptd'])

In [19]:
df_cd_2010_to_2019['DATE OCC'] = pd.to_datetime(df_cd_2010_to_2019['DATE OCC'])

In [22]:
df_cd_2010_to_2019['compare'] = np.where((df_cd_2010_to_2019['Date Rptd'] == df_cd_2010_to_2019['DATE OCC']), 'samedate', 'differentdate')

In [23]:
df_cd_2010_to_2019['compare'].value_counts()

differentdate    1124434
samedate          995363
Name: compare, dtype: int64

In [26]:
dd = df_cd_2010_to_2019[df_cd_2010_to_2019['compare'] == 'differentdate']

In [27]:
dd.head()

DR_NO  Date Rptd   DATE OCC  TIME OCC  AREA  AREA NAME  Rpt Dist No  \
1   11401303 2010-09-13 2010-09-12        45     14   Pacific         1485   
4  100100501 2010-01-03 2010-01-02      2100      1   Central          176   
5  100100506 2010-01-05 2010-01-04      1650      1   Central          162   
6  100100508 2010-01-08 2010-01-07      2005      1   Central          182   
7  100100509 2010-01-09 2010-01-08      2100      1   Central          157   

   Part 1-2  Crm Cd                                        Crm Cd Desc  ...  \
1         2     740  VANDALISM - FELONY ($400 & OVER, ALL CHURCH VA...  ...   
4         1     122                                    RAPE, ATTEMPTED  ...   
5         1     442           SHOPLIFTING - PETTY THEFT ($950 & UNDER)  ...   
6         1     330                              BURGLARY FROM VEHICLE  ...   
7         1     230     ASSAULT WITH DEADLY WEAPON, AGGRAVATED ASSAULT  ...   

    Status Desc  Crm Cd 1 Crm Cd 2 Crm Cd 3  Crm Cd 4  \
1   Invest Cont     740.0      NaN      NaN       NaN   
4   Invest Cont     122.0      NaN      NaN       NaN   
5  Adult Arrest     442.0      NaN      NaN       NaN   
6   Invest Cont     330.0      NaN      NaN       NaN   
7  Adult Arrest     230.0      NaN      NaN       NaN   

                                 LOCATION                     Cross Street  \
1         SEPULVEDA                    BL  MANCHESTER                   AV   
4         8TH                          ST  SAN PEDRO                    ST   
5  700 W  7TH                          ST                              NaN   
6         PICO                         BL  GRAND                        AV   
7  500    CROCKER                      ST                              NaN   

       LAT       LON        compare  
1  33.9599 -118.3962  differentdate  
4  34.0387 -118.2488  differentdate  
5  34.0480 -118.2577  differentdate  
6  34.0389 -118.2643  differentdate  
7  34.0435 -118.2427  differentdate  

[5 rows x 29 columns]

In [32]:
df_cd_2010_to_2019.groupby(df_cd_2010_to_2019["DATE OCC"].dt.year).count()

DR_NO  Date Rptd  DATE OCC  TIME OCC   AREA   AREA NAME  \
DATE OCC                                                             
2010      209229     209229    209229    209229  209229     209229   
2011      200792     200792    200792    200792  200792     200792   
2012      201621     201621    201621    201621  201621     201621   
2013      192689     192689    192689    192689  192689     192689   
2014      195628     195628    195628    195628  195628     195628   
2015      215641     215641    215641    215641  215641     215641   
2016      225609     225609    225609    225609  225609     225609   
2017      231303     231303    231303    231303  231303     231303   
2018      229197     229197    229197    229197  229197     229197   
2019      218088     218088    218088    218088  218088     218088   

          Rpt Dist No  Part 1-2  Crm Cd  Crm Cd Desc  ...  Status Desc  \
DATE OCC                                              ...                
2010           209229    209229  209229       209229  ...       209229   
2011           200792    200792  200792       200792  ...       200792   
2012           201621    201621  201621       201621  ...       201621   
2013           192689    192689  192689       192689  ...       192689   
2014           195628    195628  195628       195628  ...       195628   
2015           215641    215641  215641       215641  ...       215641   
2016           225609    225609  225609       225609  ...       225609   
2017           231303    231303  231303       231303  ...       231303   
2018           229197    229197  229197       229197  ...       229197   
2019           218088    218088  218088       218088  ...       218088   

          Crm Cd 1  Crm Cd 2  Crm Cd 3  Crm Cd 4  LOCATION  Cross Street  \
DATE OCC                                                                   
2010        209228     13726       226        12    209229         35613   
2011        200790     12576       231         7    200792         33584   
2012        201620     12090       193         2    201621         33206   
2013        192689     11401       177         6    192689         30868   
2014        195628     12100       206         9    195628         31476   
2015        215639     14528       415        10    215641         34909   
2016        225609     15414       482        13    225609         37246   
2017        231303     15618       506        18    231303         39681   
2018        229195     16861       609        17    229197         39541   
2019        218086     16220       496        11    218088         39471   

             LAT     LON  compare  
DATE OCC                           
2010      209229  209229   209229  
2011      200792  200792   200792  
2012      201621  201621   201621  
2013      192689  192689   192689  
2014      195628  195628   195628  
2015      215641  215641   215641  
2016      225609  225609   225609  
2017      231303  231303   231303  
2018      229197  229197   229197  
2019      218088  218088   218088  

[10 rows x 29 columns]

In [34]:
df2019 = df_cd_2010_to_2019[df_cd_2010_to_2019['DATE OCC'].dt.year == 2019]


In [35]:
df2019.shape

(218088, 29)

#### df2019 is crime data from yr 2019, 218,088 records, filtering DATE OCC 

In [42]:
df2019.to_csv(r'C:\Documents\projects\HackLA\311\data\crime data\cd2019.csv')

In [45]:
dfC2019 = pd.read_csv(r'C:\Documents\projects\HackLA\311\data\Clean_call_pp\clean_311_data_2019.csv', low_memory = False, index_col = 0)

In [49]:
dfC2019.head(2)

requestId      srnumber  councilId       councilName  typeId     typeName  \
0          1  1-1262692791         32  Greater Wilshire       4  Bulky Items   
1          2  1-1262693531          0        No council      10  Water Waste   

   agencyId                   agencyName  sourceId  sourceName  \
0         2            Sanitation Bureau        17     Website   
1         6  Department of Water & Power         7  Mobile App   

           createdDate           closedDate                          address  \
0  2019-01-01T00:02:00  2019-01-04T11:03:00         616 N GRAMERCY PL, 90004   
1  2019-01-01T00:05:00  2019-01-01T00:05:00  11500 W SAN VICENTE BLVD, 90049   

    latitude   longitude  
0  34.082258 -118.312461  
1  34.052739 -118.461185

In [50]:
df2019.head(2)

DR_NO  Date Rptd   DATE OCC  TIME OCC  AREA  AREA NAME  \
1836033  190712982 2019-06-29 2019-06-28      2200      7  Wilshire   
1836116  191910887 2019-05-24 2019-05-23      1930     19   Mission   

         Rpt Dist No  Part 1-2  Crm Cd       Crm Cd Desc  ...  Status Desc  \
1836033          702         1     510  VEHICLE - STOLEN  ...  Invest Cont   
1836116         1964         1     510  VEHICLE - STOLEN  ...  Invest Cont   

         Crm Cd 1 Crm Cd 2 Crm Cd 3  Crm Cd 4  \
1836033     510.0      NaN      NaN       NaN   
1836116     510.0      NaN      NaN       NaN   

                                        LOCATION  Cross Street      LAT  \
1836033                                   WARING        FULLER  34.0853   
1836116  9200    TOBIAS                       AV           NaN  34.2373   

              LON        compare  
1836033 -118.3498  differentdate  
1836116 -118.4516  differentdate  

[2 rows x 29 columns]

In [53]:
dfCR2019 = df2019[['DATE OCC', "TIME OCC", 'AREA NAME', 'Rpt Dist No', 'Crm Cd', 'Crm Cd Desc', 'Status Desc', 'LOCATION', 'Cross Street', 'LAT', 'LON']]
#sub = df2019[['House age', 'D. metro', 'N. shops', 'Latitude','Longitude', 'Price per unit']]

In [54]:
dfCR2019.head()

DATE OCC  TIME OCC AREA NAME  Rpt Dist No  Crm Cd       Crm Cd Desc  \
1836033 2019-06-28      2200  Wilshire          702     510  VEHICLE - STOLEN   
1836116 2019-05-23      1930   Mission         1964     510  VEHICLE - STOLEN   
1836284 2019-09-28      1800   Pacific         1491     480     BIKE - STOLEN   
1838667 2019-09-15      1600   Rampart          233     510  VEHICLE - STOLEN   
1838949 2019-05-06      1800   Mission         1999     510  VEHICLE - STOLEN   

         Status Desc                                 LOCATION Cross Street  \
1836033  Invest Cont                                   WARING       FULLER   
1836116  Invest Cont  9200    TOBIAS                       AV          NaN   
1836284  Invest Cont                   12000    VISTA DEL MAR          NaN   
1838667  Invest Cont                               OCCIDENTAL          3RD   
1838949  Invest Cont  7800    WOODMAN                      AV          NaN   

             LAT       LON  
1836033  34.0853 -118.3498  
1836116  34.2373 -118.4516  
1836284  33.9355 -118.4386  
1838667  34.0677 -118.2821  
1838949  34.2157 -118.4297

In [55]:
dfCR2019.to_csv(r'C:\Documents\projects\HackLA\311\data\WorkedonData\cr2019.csv')

In [59]:
dfCR2019['AREA NAME'].unique()

array(['Wilshire', 'Mission', 'Pacific', 'Rampart', 'Southeast', 'Newton',
       '77th Street', 'Harbor', 'Devonshire', 'Northeast', 'Southwest',
       'Van Nuys', 'West LA', 'Olympic', 'Central', 'Hollywood',
       'Hollenbeck', 'Topanga', 'Foothill', 'N Hollywood', 'West Valley'],
      dtype=object)

In [63]:
len(dfCR2019['AREA NAME'].unique())

21

In [67]:
dfC2019['councilName'].unique()

array(['Greater Wilshire', 'No council', 'Northridge West',
       'Northridge South', 'Northridge East', 'Boyle Heights',
       'Granada Hills South', 'Lincoln Heights', 'Harbor City',
       'Westside', 'Lake Balboa', 'LA-32', 'Glassell Park',
       'Hollywood Studio District', 'Elysian Valley Riverside',
       'Valley Village', 'Bel Air-Beverly Crest', 'Del Rey',
       'Mid City West', 'Wilmington', 'Historic Highland Park', 'Venice',
       'Harbor Gateway North', 'Pacoima', 'Woodland Hills-Warner Center',
       'Westlake South', 'Mission Hills', 'North Hills East',
       'Atwater Village', 'Canoga Park', 'Echo Park', 'Park Mesa Heights',
       'Wilshire Center - Koreatown', 'East Hollywood', 'Westwood',
       'Mar Vista', 'Empowerment Congress North', 'Downtown Los Angeles',
       'Sherman Oaks', 'Empowerment Congress Central', 'Van Nuys',
       'West Hills', 'West Adams', 'Panorama City', 'Reseda',
       'Central San Pedro', 'Greater Toluca Lake', 'Chatsworth',
       

In [65]:
dfCR2019.head()

DATE OCC  TIME OCC AREA NAME  Rpt Dist No  Crm Cd       Crm Cd Desc  \
1836033 2019-06-28      2200  Wilshire          702     510  VEHICLE - STOLEN   
1836116 2019-05-23      1930   Mission         1964     510  VEHICLE - STOLEN   
1836284 2019-09-28      1800   Pacific         1491     480     BIKE - STOLEN   
1838667 2019-09-15      1600   Rampart          233     510  VEHICLE - STOLEN   
1838949 2019-05-06      1800   Mission         1999     510  VEHICLE - STOLEN   

         Status Desc                                 LOCATION Cross Street  \
1836033  Invest Cont                                   WARING       FULLER   
1836116  Invest Cont  9200    TOBIAS                       AV          NaN   
1836284  Invest Cont                   12000    VISTA DEL MAR          NaN   
1838667  Invest Cont                               OCCIDENTAL          3RD   
1838949  Invest Cont  7800    WOODMAN                      AV          NaN   

             LAT       LON  
1836033  34.0853 -118.3498  
1836116  34.2373 -118.4516  
1836284  33.9355 -118.4386  
1838667  34.0677 -118.2821  
1838949  34.2157 -118.4297